### PR TITLE
Admin enables merchant#19

### LINKED ISSUE #19: User Story 49, Admin enables a merchant account
```
As an admin merchant
When I visit the merchant index page
And I click on a "enable" button for a disabled merchant
I am returned to the admin's merchant index page
And I see a flash message that the merchant's account is now enabled
And I see that the merchant's account is now enabled
This merchant can now log in
```

### Checklist:

- [ ] base code tests are written
- [ ] base code is written to pass tests
- [ ] edge case ("sad path") tests written (if required)
- [ ] edge case code written (if required)
- [ ] code is reviewed
- [ ] branch is merged
- [ ] all tests still pass

child of #24

depends on #21

### PR DESCRIPTION
**Pull request by danbriechle**
Adds enable routes 
Adds enable method
Adds current_admin? user_params 

tests passing 

take a solid look at this code its covers a few things we haven't learned yet
closes #19

### CODE REVIEW

File: app/views/admin/users/show.html.erb [hunk 1]
```diff
@@ -0,0 +1,4 @@
+<li><%= @merchant.name %></li>
```
Comment (asmolentzov) [inline app/views/admin/users/show.html.erb]: Related to above, could this just be current_user?

File: app/controllers/users_controller.rb [hunk 2]
```diff
@@ -54,9 +54,24 @@ def update
     end
   end
 
+  def enable
+    user = User.find(params[:user_id])
+    user.update(:enabled => params[:enabled])
+    if user[:enabled] = true
```
Comment (asmolentzov) [inline app/controllers/users_controller.rb]: This needs a `==` else it's assignment!

File: app/controllers/users_controller.rb [hunk 3]
```diff
@@ -54,9 +54,24 @@ def update
...
+    end
+    redirect_to "/merchants"
+  end
+
   private
 
   def user_params
-    params.require(:user).permit(:name, :street, :city, :state, :zip, :email, :password)
+    if current_admin?
+      params.fetch(:user, {}).permit(:name, :street, :city, :state, :zip, :email, :password, :enabled, :user_id)
```
Comment (asmolentzov) [inline app/controllers/users_controller.rb]: This is interesting, I'm not familiar with fetch! Note to ask you to explain how this works.

File: app/views/users/index.html.erb [hunk 4]
```diff
@@ -6,11 +6,11 @@
       <li><%= merchant.street %></li>
       <li><%= merchant.created_at %></li>
         <% if current_admin? %>
-        <li><%= link_to(admin_merchants_path(merchant.id)) %></li>
+        <li><%= link_to("#{merchant.name}", admin_merchant_path(merchant.id)) %></li>
           <% if merchant.enabled %>
-          <li><%= button_to("disable") %></li>
+          <li><%= button_to "disable", "/deactivate", method: :enable, params: { user_id: merchant.id, enabled: false } %></li>
```
Comment (asmolentzov) [inline app/views/users/index.html.erb]: Should probably be `deactivate_path`, it looks like you've got the routes worked out for that to work!

File: app/views/users/index.html.erb [hunk 5]
```diff
@@ -6,11 +6,11 @@
...
           <% else  %>
-          <li><%= button_to("enable") %></li>
+          <li><%= button_to "enable", "/activate", method: :enable, params: { user_id: merchant.id, enabled: true } %></li>
```
Comment (asmolentzov) [inline app/views/users/index.html.erb]: Ditto to above - `activate_path`

File: spec/features/admin_sees_merchant_info_spec.rb [hunk 6]
```diff
@@ -23,27 +23,84 @@
...
+
+    click_on "enable"
+
+    end
+
+    expect(page).to have_content("#{merchant_2.name} is now enabled")
+    
+    within "#merchant-#{merchant_2.id}" do
+
+    expect(page).to have_button("disable")
```
Comment (asmolentzov) [inline spec/features/admin_sees_merchant_info_spec.rb]: Would it make sense here to also check the attribute of the merchant object directly? Something like `expect(merchant_2.enabled).to eq(true)` - not exactly sure how that would work, not sure if the booleans make the checking a bit harder, but I'm sure there's an RSpec method for it.

File: app/controllers/admin/users_controller.rb [hunk 7]
```diff
@@ -4,4 +4,8 @@ def index
     @users = User.default_users
   end
 
+  def show
+    @merchant = User.find(params[:id])
```
[the remark this answers is not in the document]
Comment (Bradniedt) [reply app/controllers/admin/users_controller.rb]: I think this is looking at another user as a admin user.

[the remark this answers is not in the document]
Comment (danbriechle) [reply app/controllers/admin/users_controller.rb]: Its gotta be @merchant because its a show of merchants while the current user is the admin

File: app/controllers/users_controller.rb (hunk 2 above)
Comment (danbriechle) [reply app/controllers/users_controller.rb]: oops fixed it and still passing :)

File: app/controllers/users_controller.rb (hunk 3 above)
Comment (danbriechle) [reply app/controllers/users_controller.rb]: it was removed as it wound up being un-needed but could come in handy in the future. It was needed while i was going down a different path it enables both current user and the user object we want to edit as being permits.

File: spec/features/admin_sees_merchant_info_spec.rb (hunk 6 above)
Comment (danbriechle) [reply spec/features/admin_sees_merchant_info_spec.rb]: LMK when you find the rspec method. As discussed in person this is kind of tricky :(
